test/src/nyc-test.js

```diff
@@ -232,6 +248,39 @@ describe('nyc', function () {
       })
     })
 
+    describe('compile handlers for custom extensions are assigned', function () {
+      it('assigns a function to custom extensions', function () {
+        var nyc = new NYC({
+          cwd: process.cwd()
```

Comment:
> The reason I didn't is because the tests will use package.json by default over the opts argument. I patterned this after the config.require / opts.require that already exists.
> 
> I agree that it is not intuitive (I would expect opts to override config), but I figured the most important thing is consistency.

Oh I hadn't even realized it worked that way. Could you open an issue so @bcoe can weigh in? I suppose technically that'd be a breaking change.

---

That aside my larger concern is using the main `package.json` for these tests. It was kind of convenient for the `exclude` option but it seems strange to enable the `.es6` extension within the nyc codebase.

I suppose it's OK for now, could refactor into using a separate fixture `package.json` later.